core/src/main/scala/org/bitcoins/core/wallet/utxo/UTXOSpendingInfo.scala

```diff
@@ -536,33 +848,67 @@ case class P2WPKHV0SpendingInfo(
   require(scriptWitness.pubKey == signer.publicKey,
           "Witness has incorrect public key")
 
+  override val requiredSigs: Int = 1
+
   override def signers: Seq[Sign] = Vector(signer)
   override def conditionalPath: ConditionalPath =
     ConditionalPath.NoConditionsLeft
 }
 
-case class P2WSHV0SpendingInfo(
+sealed trait P2WSHV0SpendingInfo extends SegwitV0NativeUTXOSpendingInfoSingle {
+  require(
+    CryptoUtil
+      .sha256(scriptWitness.redeemScript.asmBytes) == scriptPubKey.scriptHash,
+    "Witness has incorrect script")
```

Comment:
nit: In general with require statements it is very nice to put the inputs into the error message so you can see what it was that failed